anycloud/cli/src/logger/mod.rs

```diff
@@ -44,15 +44,15 @@ macro_rules! error {
 
 #[macro_export]
 macro_rules! warn {
-  ($errCode:ident, $($message:tt)+) => {async{
+  ($errCode:ident, $($message:tt)+) => {
     let err_type = $crate::logger::ErrorType::$errCode;
     eprintln!($($message)+);
     $crate::deploy::client_error(err_type, &format!($($message)+), "warn").await;
-  }};
-  (metadata: $metadata:tt, $errCode:ident, $($message:tt)+) => {async{
+  };
+  (metadata: $metadata:tt, $errCode:ident, $($message:tt)+) => {
     let err_type = $crate::logger::ErrorType::$errCode;
     let value = json!($metadata);
     eprintln!($($message)+);
     $crate::deploy::client_error(err_type, &format!($($message)+), "warn").await;
```

Comment:
How does this work if you `await in a non-async function?

Reply:
You'd have to use one of `tokio`'s functions directly to achieve this. Assuming the non-async fn is called by an async fn in a tokio context, the code would look like this:
```rust
use tokio::runtime::Handle;
// before my change
Handle::current().block_on(warn!(...));
// after my change
Handle::current().block_on(async { warn!(...) });
```
neither are very convenient, but with my macro change there will be a compile error for not having an async block, whereas without it there's no compile error for not `.await`ing the future from the macro